### DISCUSSION TITLE
SoundLoc-EQ v1.0.0

### DISCUSSION THREAD

**Discussion by liincu4n** [Metacoustix/SoundLoc-EQ#5] [Announcements]
# SoundLoc-EQ v1.0.0

**Release Date**: March 14, 2025

We’re excited to announce the first official release of **SoundLoc-EQ**, a framework for unbiased sound source localization in circular spaces using deep learning. This release marks the completion of the project, delivering a robust solution for audio localization tasks.

## Key Features
- **Unbiased Encoding and Decoding**: The core `encoding_decoding.py` module introduces `unbiased_encoding` and `weighted_adjacent_decoding` to eliminate quantization errors in 360° sound localization.
- **Circular Space Support**: Handles 360-degree localization with a grid resolution of 72 cells (5 degrees each).
- **Distributed Training**: Supports multi-GPU training with PyTorch Lightning’s DDP strategy.
- **Comprehensive Evaluation**: Tested on datasets like `Libri_Circular_4` and `Libri_adhoc_nodes10_splited`, with robust performance in noisy and reverberant conditions.

## What’s New
- Initial release of the complete SoundLoc-EQ framework.
- Full implementation of training (`train.py`) and evaluation (`eval.py`) pipelines.
- Detailed documentation in `README.md`, including setup, usage, and configuration instructions.
- Pre-trained model checkpoint (`1hot_72.ckpt`) included for immediate evaluation.

## Installation
1. Clone the repository:
   ```bash
   git clone https://github.com/MetaAcoustix/SoundLoc-EQ.git
   cd SoundLoc-EQ
2. Install dependencies:
   ```bash
   pip install &&..
Follow the README.md for dataset setup and further instructions.

## Usage
- **Train**: Run `python train.py` to train the SS2net model on your dataset.
- **Evaluate**: Run `python eval.py` to test the PhaseModel with the provided checkpoint.

## Known Issues
- Dataset paths in `train.py` and `eval.py` are hardcoded (e.g., `/home/liincuan/...`). Update these to match your local environment.
- GPU support requires CUDA-compatible PyTorch; CPU fallback is available but slower.

## Future Plans
- Add a `Project Report` file for easier dependency management.

## Acknowledgments
- Thanks to the dataset providers for `Libri_Circular_4` and `Libri_adhoc_nodes10_splited`.
- Inspired by research in unbiased label distribution learning.

For more details, check the [README](https://github.com/MetaAcoustix/SoundLoc-EQ/blob/main/README.md). We welcome feedback via GitHub issues!



<hr /><em>This discussion was created from the release <a href='https://github.com/MetaAcoustix/SoundLoc-EQ/releases/tag/v1.0.0'>SoundLoc-EQ v1.0.0</a>.</em>
